Mathlib/Geometry/Manifold/ContMDiff/Defs.lean

```diff
@@ -0,0 +1,917 @@
+  have : ContDiffWithinAt 𝕜 n id (range I) (I x) := contDiff_id.contDiffAt.contDiffWithinAt
+  refine this.congr (fun y hy => ?_) ?_
+  · simp only [ModelWithCorners.right_inv I hy, mfld_simps]
+  · simp only [mfld_simps]
+#align cont_diff_within_at_prop_id contDiffWithinAtProp_id
+
+/-- A function is `n` times continuously differentiable within a set at a point in a manifold if
+it is continuous and it is `n` times continuously differentiable in this set around this point, when
+read in the preferred chart at this point. -/
+def ContMDiffWithinAt (n : ℕ∞) (f : M → M') (s : Set M) (x : M) :=
+  LiftPropWithinAt (ContDiffWithinAtProp I I' n) f s x
+#align cont_mdiff_within_at ContMDiffWithinAt
+
+/-- Abbreviation for `ContMDiffWithinAt I I' ⊤ f s x`. See also documentation for `Smooth`.
+-/
+@[reducible]
+def SmoothWithinAt (f : M → M') (s : Set M) (x : M) :=
+  ContMDiffWithinAt I I' ⊤ f s x
+#align smooth_within_at SmoothWithinAt
+
+/-- A function is `n` times continuously differentiable at a point in a manifold if
+it is continuous and it is `n` times continuously differentiable around this point, when
+read in the preferred chart at this point. -/
+def ContMDiffAt (n : ℕ∞) (f : M → M') (x : M) :=
+  ContMDiffWithinAt I I' n f univ x
+#align cont_mdiff_at ContMDiffAt
+
+theorem contMDiffAt_iff {n : ℕ∞} {f : M → M'} {x : M} :
+    ContMDiffAt I I' n f x ↔
+      ContinuousAt f x ∧
+        ContDiffWithinAt 𝕜 n (extChartAt I' (f x) ∘ f ∘ (extChartAt I x).symm) (range I)
+          (extChartAt I x x) :=
+  liftPropAt_iff.trans <| by rw [ContDiffWithinAtProp, preimage_univ, univ_inter]; rfl
+#align cont_mdiff_at_iff contMDiffAt_iff
+
+/-- Abbreviation for `ContMDiffAt I I' ⊤ f x`. See also documentation for `Smooth`. -/
+@[reducible]
+def SmoothAt (f : M → M') (x : M) :=
+  ContMDiffAt I I' ⊤ f x
+#align smooth_at SmoothAt
+
+/-- A function is `n` times continuously differentiable in a set of a manifold if it is continuous
+and, for any pair of points, it is `n` times continuously differentiable on this set in the charts
+around these points. -/
+def ContMDiffOn (n : ℕ∞) (f : M → M') (s : Set M) :=
+  ∀ x ∈ s, ContMDiffWithinAt I I' n f s x
+#align cont_mdiff_on ContMDiffOn
+
+/-- Abbreviation for `ContMDiffOn I I' ⊤ f s`. See also documentation for `Smooth`. -/
+@[reducible]
+def SmoothOn (f : M → M') (s : Set M) :=
+  ContMDiffOn I I' ⊤ f s
+#align smooth_on SmoothOn
+
+/-- A function is `n` times continuously differentiable in a manifold if it is continuous
+and, for any pair of points, it is `n` times continuously differentiable in the charts
+around these points. -/
+def ContMDiff (n : ℕ∞) (f : M → M') :=
+  ∀ x, ContMDiffAt I I' n f x
+#align cont_mdiff ContMDiff
+
+/-- Abbreviation for `ContMDiff I I' ⊤ f`.
+Short note to work with these abbreviations: a lemma of the form `cont_mdiff_foo.bar` will
+apply fine to an assumption `smooth_foo` using dot notation or normal notation.
```

Comment:
```suggestion
Short note to work with these abbreviations: a lemma of the form `ContMdiffFoo.bar` will
apply fine to an assumption `SmoothFoo` using dot notation or normal notation.
```

Reply:
I went with `ContMDiffFoo`, matching the lemmas below.